lib/bundler/cli.rb

```diff
@@ -784,7 +784,7 @@ def warn_on_outdated_bundler
       return unless SharedHelpers.md5_available?
 
       latest = Fetcher::CompactIndex.
-               new(nil, Source::Rubygems::Remote.new(URI("https://rubygems.org")), nil).
+               new(nil, Source::Rubygems::Remote.new(Bundler::URI("https://rubygems.org")), nil).
                send(:compact_index_client).
                instance_variable_get(:@cache).
                dependencies("bundler").
```

Comment:
I don't think all of these changes in `lib` are actually necessary because the `Bundler::URI` constant will beat `URI` anyways, but maybe they make it more explicit that we're using a vendored version of the `uri` gem?